lib/ddtrace/contrib/sucker_punch/instrumentation.rb

```diff
@@ -78,10 +78,8 @@ def datadog_configuration
             end
 
             def __with_instrumentation(name)
-              pin = Datadog::Pin.get_from(::SuckerPunch)
-
-              Datadog::Tracing.trace(name, service: pin.service) do |span|
-                span.span_type = pin.app_type
+              Datadog::Tracing.trace(name, service: datadog_configuration[:service_name]) do |span|
+                span.span_type = Datadog::Ext::AppTypes::WORKER
```

Comment:
Interesting -- why doesn't this one need a pin anymore?

Reply:
Pin was being used to ferry in service name in a superfluous manner. It would be initialized onto a global constant with the value from the configuration, which means it would always just reflect the configuration. There's no point in doing this.